### DISCUSSION TITLE
Is it possible to use Whisper_Mic with a fine-tuned whisper model?

### DISCUSSION THREAD

**Discussion by yilmazay74** [mallorbc/whisper_mic#58] [Q&A]
Hi Everyone,
I want to use whisper_mic with our fine tuned whisper model. However, it seems that whisper_mic is bound to be used only with whisper's own models. 
Is there any way or a workaround to make that possible ?
REgards

**Comment by yilmazay74** [comment]
Why can't we just give a fine-tuned model as a model input? I don't understand why we should to limit only whisper pre-trained models?
Would not it be nice if we could use our own custom models ?

**Comment by mallorbc** [comment]
Is this possible?  Definitely.   But not with the current code implementation. 

I have finetuned Whisper with the Huggingface implementation, but without conversion, that model does not work with OpenAI Whisper.  At the same time, the Huggingface implementation is faster.

So to make this possible, I would suggest adding support for other Whisper implementations, namely the HuggingFace one.  Then, you could change the code to allow models that are not in the list of approved models.

I will try to find time to do this myself, but I am more than open to PRs
